File: schemas/cms-blog/convex/series_items.ts
// series_items: Ordered post assignments within a content series.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const seriesItems = defineTable({
  seriesId: v.id("series"),
  postId: v.id("posts"),
  sortOrder: v.number(),
})
  .index("by_series_id_post_id", ["seriesId", "postId"])
  .index("by_post_id", ["postId"])
  .index("by_series_id_sort_order", ["seriesId", "sortOrder"]);

File: schemas/cms-blog/spacetimedb/series_items.rs
// series_items: Ordered post assignments within a content series.
// See README.md for full design rationale.

#[spacetimedb::table(name = series_items, public)]
pub struct SeriesItem {
    #[primary_key]
    pub id: String, // UUID
    pub series_id: String, // FK -> series.id (cascade delete)
    #[index(btree)]
    pub post_id: String, // FK -> posts.id (cascade delete)
    pub sort_order: i32, // Composite index: (series_id, sort_order)
    // Composite unique: (series_id, post_id)
}
